app/assets/javascripts/ui_components/modal.coffee

```diff
@@ -0,0 +1,13 @@
+$ ->
+  _.each $('.ui-components-modal'), (modal) ->
+    $modal = $ modal
+    trigger = $($modal.attr('data-trigger'))
+    return if trigger.length == 0
+    $(trigger).click (e) ->
+      url = $modal.attr('data-url')
+      if url && url.length > 0
+        $modal.find('.modal-body').load url, ->
+          $modal.one('shown.bs.modal', (e) -> $modal.trigger('uic:domchange'))
```

Comment:
Because you can't select an element that's not part of the DOM, which the remotly loaded content isn't as of `.modal('show')` returning. We could, of course, jump through more hoops, fetch the content, trigger `uic:domchange` on it and put it into the modal afterwards. But it doesn't really make things any less complex. I don't really understand why this is such a big deal.